rom/laghos.cpp

```diff
@@ -1509,7 +1509,7 @@ int main(int argc, char *argv[])
                     MFEM_VERIFY(RKStages.size() == RKStepNumSamples, "Inconsistent number of Runge Kutta stages.");
                     for (int RKidx = 0; RKidx < RKStepNumSamples; ++RKidx)
                     {
-                        sampler->SampleSolution(RKTime[RKidx], dt, RKStages[RKidx]);
+                        sampler->SampleSolution(RKTime[RKidx], last_dt, RKStages[RKidx]);
```

Comment:
I'm not sure how the sampler works in libROM (would need to review the code), but I wonder whether it matters to the sampler and its determination of when to sample that multiple samples are taken within the same timestep, with the same timestep size. That is, this is inconsistent input to the sampler. I don't think it matters for Laghos, as we end up sampling every timestep anyway, so we don't actually rely on the sampler to accurately determine when to sample. I am not saying this needs to be changed, just commenting that it is strange and perhaps should be thought about more in the future.

Reply:
Yes, I think we also have take a similar shortcut by setting an artificial time and timestep in `mergeXV`. Though the code works and every snapshots are sampled, we should review libROM and avoid the potential misuse of `dt`.